### PR TITLE
[scripts/release] Always branch off of stable.

### PR DESCRIPTION
**Pull request by jverkoey**
Prior to this change, our release process was merging `develop` directly into `stable` without first branching off of `stable`.

The result of this process was that `stable` shares no common ancestor with `develop`, making it difficult to calculate differences between a commit on `develop` and a commit on `stable`.

One example of where this becomes problematic is in generating internal CLs for our PRs and for `develop`. It is not possible to compute the delta between the most recent `stable` commit that was mirrored internally and a given `develop` commit, making it difficult for tooling to apply deltas to the internal codebase without simply blowing away the internal directory and checking out the desired `develop` commit.

To fix this problem, we need to create a common ancestor between `develop` and `stable`. To do so, when cutting a release we need to cut the `release-candidate` from `stable` rather than `develop`, and then immediately merge `develop` into the release-candidate. This creates a common ancestor between `develop` and `stable`.

Note that this change causes us to deviate from the pure git flow model we have historically been following, as shown in the before/after of the model below (note the new red lines connecting stable to develop).

| Before | After |
|:--|:--|
| git-model@2x | git-model@2x copy 2 |

### CODE REVIEW

File: scripts/release [hunk 1]
```diff
@@ -130,18 +130,18 @@ cut_release() {
...
   if $isHotFix; then
-    branch=origin/stable
-    branch_message="This is a hotfix... branching off $branch"
+    echo "This is a hotfix... branching off origin/stable and awaiting cherry-picks."
+    git checkout -b release-candidate origin/stable
   else
-    branch=origin/develop
-    branch_message="This is a normal release... branching off $branch"
+    echo "This is a normal release... branching off origin/stable and merging origin/develop"
+    git checkout -b release-candidate origin/stable
```
Comment (randallli) [inline scripts/release]: move this above the if branching to make it more clear that it is the same for both hotfix and normal release.

File: scripts/release [hunk 2]
```diff
@@ -130,18 +130,18 @@ cut_release() {
...
+    git merge --no-ff origin/develop --no-edit
```
Comment (randallli) [inline scripts/release:142]: Does this have the potential for merge conflicts? we have different .gitattributes that often gets "corrected" after.

File: scripts/release [hunk 3]
```diff
@@ -130,18 +130,18 @@ cut_release() {
...
   fi
-  echo $branch_message
-  git checkout -b release-candidate $branch
+
   git checkout origin/stable -- .gitattributes
```
Comment (randallli) [inline scripts/release:145]: does this need to be committed?

**Review by randallli — CHANGES_REQUESTED** [review]
If we go forward with this it dont forget to point out that we are deviating from the "normal" gitflow in our docs. 
I found 2 references:
https://github.com/material-components/material-components-ios/blob/c9fb2d91a4278d22da9151a8b49b546d6217e628/scripts/README-release.md
and:
https://github.com/material-components/material-components-ios/blob/172e8d7acdb1094e19ee5d5519c9bcfa16bbfbfa/contributing/releasing.md

File: scripts/release (hunk 1 above)
Comment (jverkoey) [reply scripts/release]: Done.

File: scripts/release (hunk 2 above)
Comment (jverkoey) [reply scripts/release:142]: I've just tested the commands manually locally and it does not result in a merge conflict.

File: scripts/release (hunk 3 above)
Comment (jverkoey) [reply scripts/release:145]: Not sure, but this PR isn't affecting this line so changing that behavior is out of scope.

Comment (randallli) [reply scripts/release:145]: ok cool.

File: scripts/README-release.md [hunk 4]
```diff
@@ -20,7 +20,9 @@ version tag.
 ## Cut the release
 
 When a stable release is ready to be cut, a `release-candidate` branch is `cut`
-from the latest `origin/develop` commit.
+from the latest `origin/stable` commit and `origin/develop` is merged into the
+`release-candidate`, unless the release is a hotfix in which case nothing is
+merged into the branch by default.
```
Comment (randallli) [inline scripts/README-release.md]: merged into the `release-candidate` branch in preparation for cherry picks.

File: scripts/release (hunk 2 above)
Comment (randallli) [reply scripts/release:142]: ok cool.

**Review by randallli — APPROVED** [review]

File: scripts/README-release.md (hunk 4 above)
Comment (jverkoey) [reply scripts/README-release.md]: Done.

**Pull request merged**